src/ui/public/modals/popover.js

```diff
@@ -0,0 +1,18 @@
+import angular from 'angular';
+
+/**
+ * Appends the element to the dom on instantiation, and removes it when destroy is called.
+ */
+export class Popover {
+  constructor(element) {
+    this.element = element;
+    angular.element(document.body).append(this.element);
```

Comment:
One killer improvement would be to rename this file `modal_overlay.js` and rename the class `ModalOverlay`. Then we could make this logic responsible for creating the modal overlay wrapper element:

```javascript
this.element = /* don't remember how to create an element, but that's what we should do here */;
this.element.append(element);
```